File: 백준/Silver/3986. 좋은 단어/좋은 단어.cc
#include <iostream>
#include <stack>
using namespace std;

int n, cnt;
string s;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);
    cin >> n;
    for(int i = 0; i < n; i++) {
        cin >> s;
        stack<char> st;
        for(int j = 0; j < s.size(); j++) {
            if(st.empty() || st.top() != s[j]) {
                st.push(s[j]);
            } else {
                st.pop();
            }
        }
        if(st.empty()) {
            cnt++;
        }
    }
    cout << cnt;
    return 0;
}

File: 백준/Silver/3986. 좋은 단어/좋은 단어.java
import java.util.*;

public class Main {

    private static int n, ret;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        n = sc.nextInt();
        while(n > 0) {
            n--;
            Stack<Character> st = new Stack<>();
            String s = sc.next();
            char[] ch = s.toCharArray();
            boolean flag = true;
            for(char c : ch) {
                if (st.isEmpty() || st.peek() != c) {
                    st.push(c);
                } else {
                    st.pop();
                }
            }
            if(st.empty()) {
                ret++;
            }
        }
        System.out.println(ret);
    }
}
